utils/genesis/src/lib.rs

```diff
@@ -70,6 +77,15 @@ where
     Ok(network_name)
 }
 
+pub async fn get_network_name_from_genesis_cid(cid: &str) -> Result<String, anyhow::Error> {
```

Comment:
I would not return an `Error` here, because it would make running a custom network fail. An `Option` seems more adequate to me.

Reply:
this func is not required now